### ISSUE TITLE
AttributeError: Attribute 'sym_sealed' does not exist in <class 'pyglove.core.symbolic.dict.Dict'>.

### ISSUE THREAD

**Issue by D-X-Y** [google/pyglove#191]
```
(most recent call last):
  File "/.local/lib/python3.9/site-packages/pyglove/core/symbolic/dict.py", line 628, in __setitem__
    if base.treats_as_sealed(self):
  File "/.local/lib/python3.9/site-packages/pyglove/core/symbolic/base.py", line 2386, in treats_as_sealed
    return value.sym_sealed if sealed_in_scope is None else sealed_in_scope
  File ".local/lib/python3.9/site-packages/pyglove/core/symbolic/dict.py", line 697, in __getattr__
    raise AttributeError(
AttributeError: Attribute 'sym_sealed' does not exist in <class 'pyglove.core.symbolic.dict.Dict'>.
```
Get some deeply nested errors and can not be able to figure out the root reasons.

**Comment by daiyip** [comment]
Hmm, could you give me example code for repro? Actually `sym_sealed` is defined in `pyglove.core.symbolic.Symbolic`.

**Comment by D-X-Y** [comment]
Unfortunately, I can not easily reproduce it, that I can say is that the pyglove Object that involved is a pretty simple one such as 
```
class AA(pg.Object):
    tokenizer: Annotated[AbstractTokenizer, "Tokenizer"] = get_tokenizer(
        "..."
    )

    max_num: Annotated[
        int,
        "The maximum number of tokens for prefix, where -1 means infinity.",
    ] = -1
```
The error occurs in some weird spark runs and can not easily be debugged. However, replacing pg.Object with Object works fine on the codes.

**Comment by D-X-Y** [comment]
I'm not sure whether it conflicts with some underling serialization / deserialization functions used in other libraries.

**Comment by daiyip** [comment]
Not sure. Can you share longer callstacks?

**Comment by D-X-Y** [comment]
Seems not more callstacks, it was like:
```
23/07/21 16:41:04 WARN TaskSetManager: Lost task 0.0 in stage 9.0 (TID 165) (216.153.61.120 executor 14): org.apache.spark.api.python.PythonException: Traceback (most recent call last):
  File "~/.local/lib/python3.9/site-packages/pyglove/core/symbolic/dict.py", line 628, in __setitem__
    if base.treats_as_sealed(self):
...
    ```

**Comment by D-X-Y** [comment]
I found a minimum reproducible example:

```
import pickle
from typing import Annotated, Dict
import pyglove as pg

class A(pg.Object):

  x: Annotated[Dict, "."] = dict()
  y: Annotated[str, "."] = ".."

# Define a simple dictionary
# data = A(x={1:1})
data = A()

# Serialization
serialized_data = pickle.dumps(data)
print(f"Serialized data: {serialized_data}")

# Deserialization
deserialized_data = pickle.loads(serialized_data)
print(f"Deserialized data: {deserialized_data}")
```

**Comment by daiyip** [comment]
Great, it turns out `pickle` does not work well with `pg.Symbolic` types previously.
Fixed by #194.

**Comment by D-X-Y** [comment]
Nice, thanks for your supports!
